jardin/database/base.py

```diff
@@ -19,30 +20,24 @@ def __init__(self, db_config, name, pool_config=None):
     @contextmanager
     def connection(self):
         try:
-            if self._connection is None:
-                self._connection = self.connect()
-            yield self._connection
-        except self.DRIVER.InterfaceError:
-            self._connection = None
-            raise
+            conn = self.get_connection()
+            yield conn
+            if self.autocommit:
+                conn.commit()
         except Exception as e:
             self.rollback()
-            raise
         finally:
-            self.commit()
-    
-    def commit(self):
-        if self.autocommit:
-            self._connection.commit()
             if self.pool is not None:
-                self.pool.putconn(self._connection)
```

Comment:
Could be a stupid question, @mathieuripert what is the harm if we pre-check `self._connection` is None or not. This [thread](https://stackoverflow.com/a/57851030/5862906) shed some insights on this could happen during a pool refresh (which may not be related to our error).